apptools/persistence/versioned_unpickler.py

```diff
@@ -1,5 +1,6 @@
 # Standard library imports
-from pickle import *
+from pickle import _Unpickler as Unpickler
```

Comment:
I think previously this may have been untested? `pickle.Unpickler` has no attribute `dispatch` so here I immediately see test failures with:
```
======================================================================
ERROR: test_save_load_roundtrip (apptools.persistence.tests.test_roundtrip.TestObjectSerializer)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/aayres/Desktop/apptools/apptools/persistence/tests/test_roundtrip.py", line 48, in test_save_load_roundtrip
    deserialized = serializer.load(self.tmp_file)
  File "/Users/aayres/Desktop/apptools/apptools/naming/object_serializer.py", line 58, in load
    obj = VersionedUnpickler(f).load()
  File "/Users/aayres/Desktop/apptools/apptools/persistence/versioned_unpickler.py", line 33, in load
    dispatch = self.dispatch
AttributeError: 'VersionedUnpickler' object has no attribute 'dispatch'

----------------------------------------------------------------------
```

`pickle._Unpickler` does have `dispatch`, see here: https://github.com/python/cpython/blob/aa01011003bb855cd52abfd49f2443446590d913/Lib/pickle.py#L1224

On master no tests fail (including your new one), because the logic is coming from the `VersionedUnpickler` object from `sweet_pickle` which subclasses `pickle._Unpickler` (note the classes are almost entirely copy and pasted otherwise).  



So, in summary, if we want to go on importing `pickle.Unpickler` instead of `pickle._Unpickler`, I believe we would have to rewrite the code to not use `dispatch`.  There very well be a way to do that seamlessly, it is just another change we would need to make.  The way the code is written now, it seems to assume we are using `pickle._Unpickler`?

Reply:
I see. Thanks.

Now I see that in Python 2.7, there is no such `pickle._Unpickler`, the same class was named `Unpickler` then. However in the documentation, it is clear that `dispatch` is not part of the public API, so `NewUnpickler` has depended on private and internal implementation details... which is a risky thing to do. 

For this PR should maintain behaviour of `persistence`, we'd have to use `_Unpickler` but it is clear that `dispatch` should not be depended on. Please could you open an issue to remove the dependency on private interface and reference the issue in a comment here?

Reply:
opened #232